add governance

mkorbi opened the pull request:
Signed-off-by: Max Körbächer max@koerbaecher.io

This PR adds the governance process to the WG. Will be discussed within the next meeting to be approved.
The processes and setup orientates on the TAG.

leonardpahlke commented:
/cc @leonardpahlke

rootfs commented on governance/chair-proposal-process.md:
```diff
@@ -0,0 +1,6 @@
+# Environmental Sustainability WG Chair proposal process
+
+1) Environmental Sustainability WG Co-chairs operate as a team. The Co-Chair team seeks to maintain full coverage of the leadership capabilities across the key qualities of industry experience, hands-on cloud-native and security experience, as well as administrative experience needed to run a Environmental Sustainability WG.
```
`security` -> "sustainability`

rootfs commented on governance/chair-proposal-process.md:
```diff
@@ -0,0 +1,6 @@
+# Environmental Sustainability WG Chair proposal process
+
+1) Environmental Sustainability WG Co-chairs operate as a team. The Co-Chair team seeks to maintain full coverage of the leadership capabilities across the key qualities of industry experience, hands-on cloud-native and security experience, as well as administrative experience needed to run a Environmental Sustainability WG.
+2) If a clear successor for the exiting Co-Chair exists based on the merit of their contributions ("chop wood, carry water") and relevant experience that will benefit the WG, based on the WG's collective experience over time, the exiting Co-chair may nominate a successor.
```
I don't think `chop wood`  or `carry water` is appropriately used in this context.

rootfs commented on governance/paper-process.md:
```diff
@@ -0,0 +1,235 @@
+| -- | -- | -- |
+| * [ ] | Audience, Goals, & refining scope | 1 week |
+| * [ ] | Tasking Assignment | 1 week |
+| * [ ] | Content Rough-in | 2-3 weeks |
```
I don't think `Rough-in` offers meaningful clarity in this context. Did you mean "outline" or "draft"?

rootfs commented on governance/paper-process.md:
```diff
@@ -0,0 +1,235 @@
+schedule along with the estimated time frame for completion.  Milestones are
+explained further in this document.
+
+| TODO | Milestone | Estimated time |
```
- `TODO` -> `Tasks`
- Projects can go large and small and vary among different natures, let's defer `estimated time` to a case by case basis.

rootfs commented on governance/paper-process.md:
```diff
@@ -0,0 +1,235 @@
+
+* the language origin is consistent throughout the document (lang_en or
+  lang_us),
+* phrasing is similar (caddy corner not mixed in with kitty corner),
```
The reference of `caddy corner` vs `kitty corner` is confusing. I think you mean consistent flow of the language.

mkorbi replied:
https://brians.wsu.edu/2016/05/31/caddy-corner/

rootfs replied:
Your reference only makes it more confusing: it explicitly discourages `caddy corner`. 

Back to my general comment, the goal of writing to a diverse community is to make the messaging clear. If readers have to use a dictionary, the writing is not done right.

catblade replied:
Shall we add something stating that regional slang will be, where possible, not in documents?  I am very careful when speaking across geos to avoid cultural references (not perfect, but I do my best).

I think for the above phrasing, can we just say "the language is consistent in tone throughout the document."  Honestly, I can take many different authors and do a once-over (or thrice over) and get a consistent tone, so I'm not entirely sure this is necessary with an appropriate editor.

catblade replied:
As important as an English speaker would be a non-English native speaker as a reviewer to ask on anything confusing.  English is weird and confusing.  Although this kinda argues for more people the merrier, from a clarity/understanding perspective.

rootfs commented on governance/paper-process.md:
```diff
@@ -0,0 +1,235 @@
+
+### Executive summary and content wrap-up
+
+If the paper exceeds five (5) pages, an Executive summary commensurate with the
```
Given that markdown and blogs are the prevalent forms of CNCF publications, page limit has very little meaning; it only adds confusion.

mkorbi replied:
This md exclusively talks about publishing papers, which are published as pdf. Some of these papers go 20+ pages.

catblade replied:
Let's remove anything overly prescriptive.

rootfs commented on governance/paper-process.md:
```diff
@@ -0,0 +1,235 @@
+
+### Narrative voice
+
+No more than three contributing individuals should be assigned for narrative
```
why three?

mkorbi replied:
Because with more you will never get the paper harmonized.

catblade replied:
Papers often have more than 3 contributors.  If you run it by a technical writer or have good editors, this is not an issue.  Can we place this more in terms of "narrative voice" instead of restricting contributors?

mkorbi replied:
The contributors are not limited, as well as you can have more than 3 people checking on the narrative voice.

catblade replied:
"No more than three contributing individuals" looks like you are limiting to 3.  I'm very confused by this comment.

rootfs commented on governance/paper-process.md:
```diff
@@ -0,0 +1,235 @@
+### Public comment adjudication
+
+Prior to opening the paper for community review, the group should select no more
+than 3 Adjudicators.  These individuals are tasked with accepting/rejecting
```
why 3?

mkorbi replied:
The more you have in the discussion the longer you will need to find a consensus.

rootfs replied:
The community will reach the consensus through public comment and deliberation. Vivid discussions are encouraged, not limited. Setting an arbitrary upper bound is not the way the CNCF community operates.

rootfs commented:
@mkorbi I like your timely work and appreciate your passion to build up the community. We all agree sustainability is urgent and of high impact. CNCF is doing the right thing to have this WG.

However, I disagree with certain aspects of the governance. Fundamentally, we need to
- be clear in messaging and writing. The community is diverse and around the globe. If readers need help from a dictionary to read any documents, then the writing is not done right. 
- be accommodative rather than prescriptive. In the community driven environmental, the guidelines and proposals are meant to inspire community engagement and collective wisdom.
- respect scientific methods. The paper writing process should start with scientific research and findings: what are the hypothesis, how to conduct research, what tools do we need in the research, and what findings can we infer from the research. Unfortunately, I don't see any of these in the current form of the proposal.

mkorbi commented:
> * be clear in messaging and writing. The community is diverse and around the globe. If readers need help from a dictionary to read any documents, then the writing is not done right.

It would be helpful if you could point to the wording that we have to simplify. In general, I think we will not be able to prevent this within the WG. The topic cuts through many areas, just think of clarifying what greenhouse gas means, how to measure it etc. 

> * be accommodative rather than prescriptive. In the community driven environmental, the guidelines and proposals are meant to inspire community engagement and collective wisdom.

at which part of the governance do you feel restricted or unwelcomed? The outline here is simplified of what a TAG would need to adhere, and reduced to a guidance.

> * respect scientific methods. The paper writing process should start with scientific research and findings: what are the hypothesis, how to conduct research, what tools do we need in the research, and what findings can we infer from the research. Unfortunately, I don't see any of these in the current form of the proposal.

Not every paper is scientific. The outline just should guide you through the creation process. If you have to apply scientific methods that's fully up to you.

catblade commented on README.md:
```diff
@@ -6,9 +6,27 @@ We want to raise awareness of environmental sustainability as a key element of o
 ## Meetings
 
-* The working group meets every Wednesday at 5pm CEST/GMT+2 on [Zoom](https://zoom.us/my/cncfwgenvsustainability)
+* The working group meets every Wednesday at 5pm CEST/GMT+2 
```
Can we add in PST (in addition to GMT) for US-friendly as well please?

catblade commented on governance/paper-process.md:
```diff
@@ -0,0 +1,235 @@
+scope.
+
+The paper scope and topic should be raised in at least one WG meeting to
+solicit more volunteers (ideally 4).  Interested parties should meet at least
```
I would add something in there, with volunteers, that we try to choose from a range of companies/geos/et cetera, to make sure the proposal takes into account more breadth.

catblade commented on README.md:
```diff
@@ -6,9 +6,27 @@ We want to raise awareness of environmental sustainability as a key element of o
+
+## Members
+
+> The WG is currently in its initialization phase, members needs to be elected, projects needs to be organized and roles assigned
```
*members need to be elected

-- grammar

rootfs commented:
> It would be helpful if you could point to the wording that we have to simplify. In general, I think we will not be able to prevent this within the WG. The topic cuts through many areas, just think of clarifying what greenhouse gas means, how to measure it etc.

As  commented inline in multiple places, if the readers have to use a (urban) dictionary to figure out the context, the writing loses its clarity.

> at which part of the governance do you feel restricted or unwelcomed? The outline here is simplified of what a TAG would need to adhere, and reduced to a guidance.

The CNCF community has already produced a number of well reviewed documents in each and its own ways, without sticking the prescriptive process proposed. The app delivery/security white paper writing (from which your writing process is based on) is different from e.g. serverless process. 

> Not every paper is scientific. 

This would beg the question, what the papers are for and about? 

> The outline just should guide you through the creation process. If you have to apply scientific methods that's fully up to you.

If the guideline is not projected to serve the general use, then defer it till there is more visibility.

rootfs commented on governance/tech-lead-proposal-process.md:
```diff
@@ -0,0 +1,125 @@
+
+Community nominees should adhere to the technical lead nomination requirements
+described below. Any individual who is a member of the WG list or the
+tag-security slack channel may nominate another individual.
```
`security`?

rootfs commented on governance/tech-lead-proposal-process.md:
```diff
@@ -0,0 +1,125 @@
+Nominees must:
+<!--markdownlint-disable MD005 MD007 -->
+
+* have been project or review lead on at least one project or security assessment/review
```
`security`?

wojtekidd commented on README.md:
```diff
@@ -6,9 +6,27 @@ We want to raise awareness of environmental sustainability as a key element of o
+
+> The WG is currently in its initialization phase, members need to be elected, projects needs to be organized and roles assigned
+
+### Environmental Sustainability Chairs
```
Can we add an extra chair from a different org so that we minimize any potential conflicts of interests?

leonardpahlke replied:
+1

rootfs replied:
echo that and want to put forward a team selection process so that we don't hit the scaling limit. I'd like this process based on
- Availability. The selected members should have cycles for the designated community services 
- Capability. They should be capable of doing what the community expects
- Diversity. They represent diverse backgrounds, geolocations, and affiliations.

catblade replied:
Echoing the concerns on diversity both of companies and of people.

mkorbi replied:
a quorum is intended - so ideally we will have like 3 Co-Chairs and min. 2 TLs

also would like to highlight that the TL is considered part of the leadership team

wojtekidd replied:
TL as in Team Lead?

leonardpahlke replied:
> TL as in Team Lead?

yes

rootfs replied:
I would propose 3 TLs. The TL team should sufficiently staffed to cover an entire cycle of sustainability research, practice, and end user adoption. 

- The sustainability researcher's role as a TL is to identify scientific research, methodology, and future directions. I would nominate @wangchen615 for this role.
- The practitioner's role as a TL is to use these research and methodology to integrate with CNCF ecosystem and projects. The practitioner must have track record of CNCF contribution and engagement. I would recommend @williamcaban
- The end user's role as a TL is to adopt the projects and best practices, provide feedback for enhancement and directions.

mkorbi replied:
> > TL as in Team Lead?
> 
> yes

Actually Tech Lead

rootfs commented on governance/chair-proposal-process.md:
```diff
@@ -0,0 +1,148 @@
+* Three Co-Chairs
+* Three Tech Leads
+
+As WG we strive to be a gender, company and geolocation, diverse group. 
```
It is a good start to have diversity explicitly spelled out in the first place.

catblade replied:
"as a TAG we strive..."

rootfs commented on governance/chair-proposal-process.md:
```diff
@@ -0,0 +1,148 @@
+## Environmental Sustainability WG Chair proposal process
+
+1) Environmental Sustainability WG Co-chairs operate as a team. The Co-Chair team seeks to maintain full coverage of the leadership capabilities across the key qualities of industry experience, hands-on cloud-native and sustainability experience, as well as administrative experience needed to run a Environmental Sustainability WG. 
+2) If a clear successor for the exiting Co-Chair exists based on the merit of their contributions ("plant trees, carry water") and relevant experience that will benefit the WG, based on the WG's collective experience over time, the exiting Co-chair may nominate a successor.
```
> "plant trees, carry water"

As mentioned earlier, please use the right language, otherwise they could introduce unwelcome surprises.

cdeliaRH replied:
Omit (plant trees, carry water). It's actually "chop wood, carry water" but is unnecessary and flippant in a governance file.

mkorbi commented on governance/chair-proposal-process.md:
```diff
@@ -0,0 +1,148 @@
+* Three Tech Leads
+
+As WG we strive to be a gender, company and geolocation, diverse group. 
+Therefore, the whole leadership team is not allowed to have more than two people from the same organization. This also includes companies that belong through majority ownership to another company. 
```
As you mentioned in the last meeting we should try to do it better. 

Let's say we have two chairs from Company A and two tech leads from Company B, where Company A owns a significant part of Company B, this should be considered the same organization. This shouldn't be allowed, because it doesn't guarantee diversity regards company representation. Or in another word, the two chairs and the two tech leads can always get a majority in a vote.

rootfs replied:
Why this is not allowed? Why doesn't it guarantee diversity, and in which way? It is not only discriminative to evaluate Open Source contributors based on their employments, it is full of fallacy as well. What do you do when people change jobs, go through company acquisitions, or conduct consulting projects?

Successful Open Source communities guarantee fairness and diversity through transparency and openness, not through discriminative rules.

catblade replied:
@mkorbi  it really depends on the size of the company.  I can see a small company having little diversity, but companies that work across Geos have much diversity, many different teams, and many of these have not only different goals, but different cultures.

scottrigby replied:
As an alternative solution with precedence, this is addressed by organization voting caps in the governance of projects such as [Thanos](https://github.com/thanos-io/thanos/blob/main/docs/governance.md#governance-1), [CNI](https://www.cni.dev/docs/governance/), and [Cortex](https://github.com/cortexproject/cortex/blob/master/GOVERNANCE.md).

mkorbi commented on README.md:
```diff
@@ -6,9 +6,31 @@ We want to raise awareness of environmental sustainability as a key element of o
+## Members
+
+> The WG is currently in its initialization phase, members need to be elected, projects needs to be organized and roles assigned
+
+### Environmental Sustainability Chairs
+
+* Max Körbächer ([@mkorbi](https://github.com/mkorbi)), Liquid Reply [Chair term: 01.06.2022 - 01.06.2024]
+* Leonard Pahlke ([@leonardpahlke](https://github.com/leonardpahlke)), Liquid Reply [Chair term: 01.06.2022 - 01.06.2024]
```
> AFAIK there are no current co-chairs. Why are Max and Leonard added to this PR as such?

Like with other proposed groups, the initiator gets this responsibility. I know not every WG has chairs, complex ones like the CNF do so. 

> Max thank you for sending that email. However, I think we should extend the proposed nomination period there (I will note that on the email list). For now, let's at least fix this PR and connect with CNCF on it, as the process so far is a bit atypical. We'll all get it together as a group 😄

As answered already in the mailing list, that is not correct. At the moment there is no nomination going on. It was proposed to postpone this for some weeks because the discussed/nominated people in the meeting are basically unknown, haven't contributed yet on any of the points and were a little too fast forward.

leonardpahlke commented on governance/paper-process.md:
```diff
@@ -0,0 +1,236 @@
+
+### Narrative voice
+
+No more than three contributing individuals should be assigned for narrative
+voice. The narrative voice is a semi-final pass of the paper to ensure it reads
```
```suggestion
Three contributing individuals should be assigned for narrative voice.
If there is a special need to assign more, it is possible, but should be discussed in the TAG beforehand.
The narrative voice is a semi-final pass of the paper to ensure it reads
```

catblade replied:
Please remove this entirely.  It's too prescriptive.  Narrative voice can be done with a single reviewer after the fact and as we discussed, landscape is more than 3.

leonardpahlke replied:
+1

leonardpahlke replied:
...updated, suggestion

leonardpahlke reviewed (changes requested):
I added some comments to get this PR hopefully ready to be merged soon! :) 

IMO todo:
* Update `WG` references to `TAG`
* For readability it could help to write role and team names Capitalised (for example `technical lead` -> `Technical Lead` also inside sentences)
* Line breaks are not done properly. This makes it more difficult to read.

catblade commented on governance/chair-proposal-process.md:
```diff
@@ -0,0 +1,148 @@
+
+The leadership team is conducted of:
+* Three Co-Chairs
+* Three Tech Leads
```
Let's not limit tech leads, if we can avoid?  We will have a lot of individual areas, so I can forsee a bigger team than other TAGs.

Maybe just use something more generic:

"A set of tech leads, not more than 5"?

leonardpahlke replied:
AFAIK, Project Leads do not have to be Technical Leads (of the TAG), so we are not limited to a maximum number of projects with 3 TL.

catblade replied:
I'd propose we define a set number then instead of "up to three".  Three is a good number.  :-)

amye replied:
Tech Leads aren't limited, I'd strike this overall.

catblade commented on governance/README.md:
```diff
@@ -0,0 +1,7 @@
+Environmental Sustainability Working Group is a CNCF Working Group and reports to the Technical Oversight Comittee (TOC).
+
+* [Charter](../charter.md) - mission and scope
+* [Roles](roles.md) - the work of the group is facilitated by Chairs, Technical Leads, and active group members
```
Sounds great.  Oh, do we want another one for "Media Lead" or something particular for handling conferences and keeping us organized?

leonardpahlke replied:
How about if the media lead is a maintainer (role maintainer)? Alsom, It might be useful to define further maintainer "specifics", e.g. for blog posts / articles, Github governance, general collaboration

amye replied:
From the 11/2 meeting: suggest a Tech Lead for Outreach, TAGs do not have maintainers :)

catblade commented on governance/chair-proposal-process.md:
```diff
@@ -0,0 +1,148 @@
+# Environmental Sustainability Leadership Team
+
+The leadership team of the WG are servant leads. They are an enabler for the group and support the creation and success of the different activities. They are the driver for the group for not losing the higher targets out of sight and taking decisions when no consensus can be found.
```
Move WG to TAG

catblade commented on governance/chair-proposal-process.md:
```diff
@@ -0,0 +1,148 @@
+
+1) Environmental Sustainability WG Co-chairs operate as a team. The Co-Chair team seeks to maintain full coverage of the leadership capabilities across the key qualities of industry experience, hands-on cloud-native and sustainability experience, as well as administrative experience needed to run a Environmental Sustainability WG. 
+2) If a clear successor for the exiting Co-Chair exists based on the merit of their contributions ("plant trees, carry water") and relevant experience that will benefit the WG, based on the WG's collective experience over time, the exiting Co-chair may nominate a successor.
+3) The exiting Co-Chair should propose the desired successor to the other Co-Chairs. If there are any objections, the Co-Chair will have the opportunity to propose another worthy candidate. If Environmental Sustainability WG Chairs, Tech Leads and TOC Liaisons are unable to put forward a candidate, then a selection process will need to be agreed upon and executed.
```
Another case of WG

amye replied:
I would strike this as well as this is likely to be part of TAG governance from the TOC

amye commented on governance/roles.md:
```diff
@@ -0,0 +1,259 @@
+> As WG we search for a deep alignment with the TOC and community, therefore, we apply TAG rules/processes
+
+The [CNCF TAG](https://github.com/cncf/toc/blob/master/tags) process identifies
+a TOC Liaison.  The WG Chairs are responsible for establishing effective
```
No, the TOC Liason is a member of TOC who works with you all to help prioritize. Just strike this section. :)

mkorbi commented:
I updated the gov according the comments and added the missing peaces based on the Security TAG Gov where suitable. WG should be cleaned out in the entire repo now.

mkorbi commented:
Removed presentation template for clean cut to this PR https://github.com/cncf/tag-env-sustainability/pull/47

leonardpahlke reviewed (approved):
LGTM
